src/Visualization/Python/GenerateTetrahedralConnectivity.py

```diff
@@ -0,0 +1,208 @@
+    can be used to fill gaps between finite-difference or Gauss elements.
+
+    If this algorithm is too slow, one possible improvement is to apply
+    qhull to each block of the domain and then connect the blocks to each
+    other separately. This keeps the number of grid points lower for each
+    invocation of qhull, which likely reduces total runtime and may also
+    reduce or eliminate failure cases. In the ideal case we would apply
+    qhull to each element and then connect elements that are using FD or
+    Gauss points to their neighbors.
+
+    \f
+    Arguments:
+      h5files: List of H5 volume data files.
+      subfile_name: Volume data subfile in the H5 files.
+      start_time: Optional. The earliest time at which to start visualizing. The
+        start-time value is included.
+      stop_time: Optional. The time at which to stop visualizing. The stop-time
+        value is not included.
+      stride: Optional. View only every stride'th time step.
+      coordinates: Optional. Name of coordinates dataset. Default:
+        "InertialCoordinates".
+      delete_tetrahedral_connectivity: Optional. Delete the dataset and return.
+        Default: False
+    """
+    h5files = [(h5py.File(filename, "a"), filename) for filename in h5files]
+
+    if not subfile_name:
+        subfiles = available_subfiles(
+            (h5file for h5file, _ in h5files), extension=".vol"
+        )
+        raise RequiredChoiceError(
+            (
+                "Specify '--subfile-name' / '-d' to select a"
+                " subfile containing volume data."
+            ),
+            choices=subfiles,
+        )
+
+    if not subfile_name.endswith(".vol"):
+        subfile_name += ".vol"
+
+    for h5file, filename in h5files:
+        # Open subfile
+        try:
+            vol_subfile = h5file[subfile_name]
+        except KeyError as err:
+            raise ValueError(
+                f"Could not open subfile name '{subfile_name}' in '{filename}'."
+                " Available subfiles: "
+                + str(available_subfiles(h5file, extension=".vol"))
+            ) from err
+
+        # Sort timesteps by time
+        temporal_ids_and_values = sorted(
+            [
+                (key, vol_subfile[key].attrs["observation_value"])
+                for key in vol_subfile.keys()
+            ],
+            key=lambda key_and_time: key_and_time[1],
+        )
+
+        # Stride through timesteps
+        for temporal_id, time in temporal_ids_and_values[::stride]:
+            # Filter by start and end time
+            if start_time is not None and time < start_time:
+                continue
+            if stop_time is not None and time > stop_time:
+                break
+
+            data_at_id = vol_subfile[temporal_id]
+
+            if delete_tetrahedral_connectivity:
+                if "tetrahedral_connectivity" in data_at_id:
+                    del data_at_id["tetrahedral_connectivity"]
```

Comment:
Note that to free up the memory in the h5 file for this deleted dataset you actually have to repack the h5 file (depending on the [file space strategy](https://docs.hdfgroup.org/hdf5/rfc/FileSpaceManagement.pdf)). This is done in `DeleteSubfiles.py`. So I suggest you name this option `force` and just delete the `continue` in the next line. And then later we can extend `DeleteSubfiles.py` to take a glob pattern of dataset names so it can clean up datasets in multiple observations.